File: android/src/main/java/expo/module/signature/SignaturePrompt.kt
package expo.module.signature

import androidx.biometric.BiometricManager
import androidx.biometric.BiometricPrompt
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class SignaturePrompt(
    @Field val title: String?,
    @Field val subtitle: String?,
    @Field val cancel: String?,
): Record {
    fun getPromptInfo(): BiometricPrompt.PromptInfo = BiometricPrompt.PromptInfo.Builder().run {
        title?.let { setTitle(it) }
        subtitle?.let { setSubtitle(it) }
        cancel?.let { setNegativeButtonText(it) }
        setAllowedAuthenticators(BiometricManager.Authenticators.BIOMETRIC_STRONG)
        build()
    }
}

File: ios/Module/models/SignaturePrompt.swift
//
//  SignatureInfo.swift
//  SignatureModule
//
//  Created by Riccardo Pizzoni on 05/07/24.
//

import ExpoModulesCore

struct SignaturePrompt: Record {
    @Field var title: String?
    @Field var subtitle: String?
    @Field var cancel: String?
}
